src/components/Trade/hooks/useBestSwapper.tsx

```diff
@@ -0,0 +1,43 @@
+import type { Asset } from '@shapeshiftoss/asset-service'
+import type { ChainId } from '@shapeshiftoss/caip'
+import type { Swapper } from '@shapeshiftoss/swapper'
+import { useEffect, useState } from 'react'
+import { getSwapperManager } from 'components/Trade/hooks/useSwapper/swapperManager'
+import { useTradeQuoteService } from 'components/Trade/hooks/useTradeQuoteService'
+import { getBestSwapperApi } from 'state/apis/swapper/getBestSwapperApi'
+import { selectFeatureFlags } from 'state/slices/preferencesSlice/selectors'
+import { useAppDispatch, useAppSelector } from 'state/store'
+
+type BestSwapperArgs = { feeAsset: Asset | undefined }
+
+// A helper hook to get the best swapper from the RTK API
+export const useBestSwapper = ({ feeAsset }: BestSwapperArgs) => {
```

Comment:
Two things we should be very careful about here is
- the hook deps will only invalidate if `feeAsset` changes - which could make the best swapper effectively not the *best* swapper
- swappers could be swapped mid-flow when e.g consumed in `<TradeConfirm />`, since hooks are not singletons and this would call a new hook instance, returning a new, possibly different best swapper.

Given this is consuming an RTK endpoint, which would be cached with RTK defaults(60,000 ms), we should theoretically be fine WRT to the latter, but the swapper not being the actual best is an actual concern